lib/host-resolve.c

```diff
@@ -36,6 +36,10 @@
 #include <string.h>
 #include <resolv.h>
 
+#ifndef AI_V4MAPPED
```

Comment:
I would move this to a "global" level, as in other places in the future we might relay on this constant and that would prevent breaking openbsd again.
By global I mean the `syslog-ng-config.h` which is created by configure and cmake. Also that way it is detected

Reply:
Do you mean adding a configure/cmake test and then something like this in `syslog-ng-config.h`?
```
#ifndef SYSLOG_NG_HAVE_AI_V4MAPPED
#define AI_V4MAPPED 0
#endif
```

Reply:
Unfortunately, I don't see a way to accomplish that in `syslog-ng-config.h` itself using autoconf since any define will be prefixed with `SYSLOG_NG_`.  This is the case even when using `AH_BOTTOM`

Reply:
Yes, and that needs to be remapped. But I am fine with this version. The plan is to include this change into the next release.